File: parking_system_java/Ticket.java
package parking_system;

import java.time.LocalDateTime;

class Ticket {
    private final Vehicle vehicle;
    private final String ticketId;
    public final LocalDateTime entryTime;
    public final int hourlyRate;
    public Spot spot;

    public Ticket(Vehicle v, String ticketId, LocalDateTime time, Spot spot, int rate) {
        this.vehicle = v; this.ticketId = ticketId; this.entryTime = time; this.spot = spot; this.hourlyRate = rate;
    }

    @Override
    public String toString() {
        return "Ticket{" +
                "vehicle=" + vehicle +
                ", ticketId='" + ticketId + '\'' +
                ", entryTime=" + entryTime +
                ", spot=" + spot +
                ", hourlyRate=" + hourlyRate +
                '}';
    }
}

File: cpp/parking_system/ticket.cpp
#include "ticket.h"
#include <ctime>
#include <string>

using namespace std;
using namespace std::chrono;

Ticket::Ticket(Vehicle* v, const string& id, system_clock::time_point time, Spot* s, int rate)
    : vehicle(v), ticketId(id), entryTime(time), spot(s), hourlyRate(rate) {}

string Ticket::to_string() const {
    time_t et = system_clock::to_time_t(entryTime);
    char* timeStr = ctime(&et);
    string entryTimeStr = timeStr ? string(timeStr) : "UNKNOWN_TIME";

    return "Ticket{vehicle=" + (vehicle ? vehicle->to_string() : "null") +
           ", ticketId='" + ticketId +
           "', entryTime=" + entryTimeStr +
           ", spot=" + (spot ? spot->to_string() : "null") +
           ", hourlyRate=" + std::to_string(hourlyRate) + "}";
}
